sqladmin/forms.py

```diff
@@ -23,37 +35,63 @@
 )
 from wtforms.fields.core import UnboundField
 
+from sqladmin.exceptions import NoConverterFound
 from sqladmin.fields import JSONField, QuerySelectField, QuerySelectMultipleField
 
 
+class Validator(Protocol):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        ...  # pragma: no cover
+
+    def __call__(self, form: Form, field: Field) -> None:
+        ...  # pragma: no cover
+
+
+class ConverterCallback(Protocol):
+    def __call__(
+        self,
+        model: type,
+        prop: Union[ColumnProperty, RelationshipProperty],
+        kwargs: Dict[str, Any],
+    ) -> UnboundField:
+        ...  # pragma: no cover
+
+
+T_CC = TypeVar("T_CC", bound=ConverterCallback)
+
+
 @no_type_check
-def converts(*args: str) -> Callable:
-    def _inner(func: Callable) -> Callable:
+def converts(*args: str) -> Callable[[T_CC], T_CC]:
+    def _inner(func: T_CC) -> T_CC:
         func._converter_for = frozenset(args)
         return func
 
     return _inner
 
 
 class ModelConverterBase:
-    _convert_for = None
+    _callbacks: Dict[str, ConverterCallback] = {}
 
     def __init__(self) -> None:
-        converters = {}
+        super().__init__()
+        self._register_callbacks()
+
+    def _register_callbacks(self):
+        callbacks = {}
 
         for name in dir(self):
             obj = getattr(self, name)
             if hasattr(obj, "_converter_for"):
                 for classname in obj._converter_for:
-                    converters[classname] = obj
+                    callbacks[classname] = obj
 
-        self.converters = converters
+        self._callbacks = callbacks
 
-    def get_converter(
-        self, prop: Union[ColumnProperty, RelationshipProperty]
-    ) -> Callable:
+    def get_callback(
```

Comment:
What's the reason for renaming converter to callback? Is it unclear?

Reply:
Right now "converter" refers to both the name of the class and the name of the methods within the class. The name "converter" currently does two things. But by distinguishing between converters (the class) and the _callbacks_ (the methods), we have 1 name for 1 concept each, rather than 1 name referring to 2 concepts.

We can potentially get the code to the point where we can expose this part of the API, and I think it is appropriate, if you want to do that, to use good and precise names.

Reply:
One thing I should have considered-- is "callback" even the right terminology? I think a callback is more like, a function passed in an argument. It's tenuously a callback because it's decorated. So, I'll revert it back to `converters` and `get_converter`. And I will rename `ConverterCallback` to `ConverterCallable` for the protoocol.

Reply:
Yes I agree, this is a pretty old approach, I honestly took it from WTForms and didn't go back to re-factoring it.